### PR TITLE
parameter masking

### PR DESCRIPTION
**Pull request by haakon-e**
# Purpose
This PR is intended to allow for simple functional dependencies between parameters to be calibrated. The primary use cases are the following:
1. Given a parameter vector, this PR allows you to calibrate a subset of its components, keeping the other components to their default values.
2. Define an equality relation between two parameters, either scalar parameters or components of vector parameters.

# Implementation details
In practice, to handle this, the struct `ParameterMap` is introduced:
https://github.com/CliMA/CalibrateEDMF.jl/blob/fa654dd8911895bbe4b08559b635f420270c4261/src/HelperFuncs.jl#L101-L103
The struct contains `mapping`, which is a dictionary that defines how to specify values of parameters that aren't explicitly a part of the calibration process. The dictionary keys are these parameters, while the values can either be 
1. a number. In this case, the parameter is kept fixed at the specified number.
2. a string. The string is assumed to be a parameter that is being calibrated. In this case, the parameter is set to whichever value that parameter takes on for each simulation.

Note that vector parameter components are specified directly by index, i.e., to consider the third component of the vector parameter `param`, write `param_{3}`.

For convenience, the function `do_nothing_param_map ` is also exported, which is a do-nothing `ParameterMap`, defined simply as
```julia
do_nothing_param_map() = ParameterMap(Dict())
```

# Checks:
- [x] Unit tests are added for the new functionality
- [x] Appropriate documentation is added to describe usage, including docstrings.

### CODE REVIEW

**Review by ilopezgp — COMMENTED** [review]
I think the idea of superseding dictionaries is going to complicate the logic of the pipeline.

The idea of parameter masking is useful, but I think they should be applied to parameters not included in the calibration priors. For the uses that you envision, we should have:

- A way of defining the prior for parameter components instead of the full parameter vector. Now you are defining the vector and taking out some components, defining artificial priors that are never used. Let's be more explicit and define the parameter components we want to calibrate instead. This does not require a new mask structure.
- The functional relationship mask is very useful, although I would give it a different name. In this case, we should have a similar struct to what you defined, but the mapping should be `y=f(x)`, where `x` are parameters to be calibrated and `y` are other parameters that are not in the calibration priors.

The implementation I propose doesn't have superseding dictionaries or the need to define prior information that is never used.

File: src/HelperFuncs.jl [hunk 1]
```diff
@@ -37,6 +41,160 @@ using Glob
...
+"""
+    struct Mask
+
+`Mask` is a struct that defines relations between calibrated and non-calibrated parameters.
+
+The dictionary `map_params` specifies a map from any parameters that is not being 
+calibrated to either a calibrated parameter, or a number. 
+If another parameter name is specified, the samples from that parameter's distribution
+is used as the given parameter's value throughout calibration, effectively defining
+an equivalence between the two parameters.
```
[the remark this answers is not in the document]
Comment (haakon-e) [reply src/HelperFuncs.jl:53]: I don't see how. The identity mask is supposed to do nothing

File: src/HelperFuncs.jl [hunk 2]
```diff
@@ -37,6 +41,160 @@ using Glob
...
+# ...
+# end
+```
+
+"""
+Base.@kwdef struct Mask
+    map_params::Dict
+end
+
+""" Mask that does not modify the parameters to be calibrated"""
```
[the remark this answers is not in the document]
Comment (haakon-e) [reply src/HelperFuncs.jl:106]: changed description to
```
""" do-nothing mask."""
```

File: src/HelperFuncs.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (ilopezgp) [reply src/HelperFuncs.jl:53]: As defined now, the identity mask does nothing, but I think that is a bit misleading. If you use a mask x = y (which is the only mask implemented), that could also be called an identity mask.

**Pull request merged**